nntrainer/layers/conv2d_layer.cpp

```diff
@@ -440,16 +441,12 @@ void Conv2DLayer::calcDerivative(RunLayerContext &context) {
   /// filter_kernel^T X derivaitive  -> column matrix
   /// col2im(column matrix) to reconstruct the original image
 
-  unsigned int num_threads = NNTR_NUM_THREADS;
-
-  if (num_threads > derivative.batch())
-    num_threads = 1;
-
-  if (num_threads > 1) {
+  if (NNTR_NUM_THREADS > 1) {
     auto dowork = [&](size_t s, size_t e, void *user_data) {
+      Tensor result =
```

Comment:
I think temporary tensor result makes it difficult to track actual memory usage in conv2d layer.
AFAIK We already allocate the tensor inter_result for this purpose (Maybe it needs to increase tensor size by num_threads times). How about using this variable?

Reply:
each thread needs its' own tensor. inter_results  is used at calcGradient.

Reply:
Discussed offline.
number of threads might be vary even the model is compiled so each threads needs to use temporarily allocated tensor